### ISSUE TITLE
GET Request via browser URL: "Unauthenticated User - http: named cookie not present / WebID-TLS error: No client certificate found in the TLS request!"

### ISSUE THREAD

**Issue by ghost** [jolocom/webid-proxy#6]
So I'm using the webid proxy webinterface to login. I get a pop up that says "success". This should create a cookie and allow me to make requests via the browser as my signed in user "max".

Yet, when I try to access `http://localhost:8111/proxy/https://localhost:8443/max/profile/card#me` I get:

```
------ New GET request from 172.17.0.1:42346 ------
[debug] 2016/05/02 14:33:25 auth.go:24: userCookie error: http: named cookie not present
[debug] 2016/05/02 14:33:25 auth.go:44: WebID-TLS error: Get https://max.localhost:8443/profile/card: dial tcp: lookup max.localhost on 192.168.178.1:53: no such host
[debug] 2016/05/02 14:33:25 auth.go:53: Unauthenticated User
[debug] 2016/05/02 14:33:25 server.go:264: 172.17.0.1:42346 requested resource URI: /max/profile/card
[debug] 2016/05/02 14:33:25 server.go:265: 172.17.0.1:42346 requested resource Path: /data/max/profile/card
[debug] 2016/05/02 14:33:25 acl.go:48: Checking accessTo <Read> to https://localhost:8443/max/profile/card for WebID: 
[debug] 2016/05/02 14:33:25 acl.go:49: Looking for policies in /data/max/profile/card,acl
[debug] 2016/05/02 14:33:25 acl.go:54: Found policies in /data/max/profile/card,acl
[debug] 2016/05/02 14:33:25 acl.go:98: Found accessTo policy for <Read>
[debug] 2016/05/02 14:33:25 acl.go:112: No origin found, moving on
[debug] 2016/05/02 14:33:25 acl.go:98: Found accessTo policy for <Read>
[debug] 2016/05/02 14:33:25 acl.go:112: No origin found, moving on
[debug] 2016/05/02 14:33:25 acl.go:135: Found agentClass policy
[debug] 2016/05/02 14:33:25 acl.go:137: Read access allowed as FOAF Agent
[debug] 2016/05/02 14:33:25 server.go:595: Setting CType to: text/plain
```

Looks like it's trying to get the profile file from the wrong path, i.e. `https://max.localhost:8443/profile/card` instead of `https://localhost:8443/max/profile/card
`

@peacekeeper Should we adjust the web id proxy, or do we have to fix the gold server to host the profile under the expected URL?

**Comment by ghost** [comment]
Similar things with curl, except here  we get a **WebID-TLS error: No client certificate found in the TLS request**!

```
→ curl -c cookie -v --data "username=max&password=test" http://localhost:8111/login

...

→ curl -b cookie -v http://localhost:8111/proxy/https://localhost:8443
*   Trying ::1...
* Connected to localhost (::1) port 8111 (#0)
> GET /proxy/https://localhost:8443 HTTP/1.1
> Host: localhost:8111
> User-Agent: curl/7.45.0
> Accept: */*
> Cookie: JSESSIONID=14gxs13vne7gypw6e0dz5lo5m
>
< HTTP/1.1 200 OK
< Date: Tue, 10 May 2016 10:24:02 GMT
< Accept-Patch: application/json, application/sparql-update
< Accept-Post: text/turtle, application/json
< Access-Control-Allow-Credentials: true
< Access-Control-Expose-Headers: User, Location, Link, Vary, Last-Modified, WWW-Authenticate, Content-Length, Accept-Patch, Accept-Post, Allow, Updates-Via, Ms-Author-Via
< Access-Control-Max-Age: 1728000
< Allow: OPTIONS, HEAD, GET, PATCH, POST, PUT, MKCOL, DELETE, COPY, MOVE, LOCK, UNLOCK
< Content-Type: text/turtle
< ETag: "d908beca1f19e567fb9311734ddd7ac1"
< Link: <https://localhost:8443/,acl>; rel="acl", <https://localhost:8443/,meta>; rel="meta"
< Link: <http://www.w3.org/ns/ldp#BasicContainer>; rel="type"
< Link: <http://www.w3.org/ns/ldp#Resource>; rel="type"
< Link: <https://localhost:8443/,meta>; rel="meta"
< Ms-Author-Via: DAV, SPARQL
< Updates-Via: wss://localhost:8443/
< Date: Tue, 10 May 2016 10:24:02 GMT
< Content-Length: 4587
< Server: Jetty(9.3.0.M1)
<
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .

...

* Connection #0 to host localhost left intact
```

Resulting in the following output on the Gold server:

```
[debug] 2016/05/10 10:43:30 server.go:225: 
------ New GET request from 172.17.0.1:41208 ------
[debug] 2016/05/10 10:43:30 auth.go:24: userCookie error: http: named cookie not present
[debug] 2016/05/10 10:43:30 auth.go:44: WebID-TLS error: No client certificate found in the TLS request!
[debug] 2016/05/10 10:43:30 auth.go:53: Unauthenticated User
[debug] 2016/05/10 10:43:30 server.go:264: 172.17.0.1:41208 requested resource URI: /
[debug] 2016/05/10 10:43:30 server.go:265: 172.17.0.1:41208 requested resource Path: /data/
[debug] 2016/05/10 10:43:30 acl.go:48: Checking accessTo <Read> to https://localhost:8443/ for WebID: 
[debug] 2016/05/10 10:43:30 acl.go:49: Looking for policies in /data/,acl
[debug] 2016/05/10 10:43:30 acl.go:48: Checking defaultForNew <Read> to https://localhost:8443/ for WebID: 
[debug] 2016/05/10 10:43:30 acl.go:49: Looking for policies in /data/,acl
[debug] 2016/05/10 10:43:30 acl.go:183: No ACL policies present - access allowed
```

**Comment by fgro** [comment]
Hi @peacekeeper , I think the proxy needs to set the correct Accept header to return proper html/javascript.

Here is an example:

```
→ curl -H "Accept: text/html, application/xhtml+xml, application/xml;q=0.9,*/*;q=0.8" -v -k https://localhost:8443
* Rebuilt URL to: https://localhost:8443/
*   Trying ::1...
* connect to ::1 port 8443 failed: Connection refused
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 8443 (#0)
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.2 (OUT), TLS header, Certificate Status (22):
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Request CERT (13):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Certificate (11):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN, server accepted to use http/1.1
* Server certificate:
*        subject: O=Acme Co
*        start date: Jan 30 23:52:19 2014 GMT
*        expire date: Jan 28 23:52:19 2024 GMT
*        issuer: O=Acme Co
*        SSL certificate verify result: unable to get local issuer certificate (20), continuing anyway.
> GET / HTTP/1.1
> Host: localhost:8443
> User-Agent: curl/7.45.0
> Accept: text/html, application/xhtml+xml, application/xml;q=0.9,*/*;q=0.8
>
< HTTP/1.1 200 OK
< Accept-Patch: application/json, application/sparql-update
< Accept-Post: text/turtle, application/json
< Access-Control-Allow-Credentials: true
< Access-Control-Expose-Headers: User, Location, Link, Vary, Last-Modified, WWW-Authenticate, Content-Length, Accept-Patch, Accept-Post, Allow, Updates-Via, Ms-Author-Via
< Access-Control-Max-Age: 1728000
< Allow: OPTIONS, HEAD, GET, PATCH, POST, PUT, MKCOL, DELETE, COPY, MOVE, LOCK, UNLOCK
< Content-Type: text/html
< Etag: "d908beca1f19e567fb9311734ddd7ac1"
< Link: <https://localhost:8443/index.html,meta>; rel="meta", <https://localhost:8443/index.html,acl>; rel="acl"
< Ms-Author-Via: DAV, SPARQL
< Updates-Via: wss://localhost:8443/
< User:
< Date: Tue, 10 May 2016 12:26:49 GMT
< Content-Length: 1276
<
<!DOCTYPE html>

<html>
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport"
          content="width=device-width, initial-scale=1, user-scalable=no">

    <title>Little Sister</title>

    <link href="https://fonts.googleapis.com/css?family=Roboto:400,300,600"
          rel="stylesheet"
          type="text/css" />
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons"
          rel="stylesheet">

    <style type="text/css">
      * {
        margin: 0;
        padding: 0;
      }
      html, body {
        height: 100%;
      }
      body {
        font-family: Roboto;
      }
      #app {
        width: 100%;
        height: 100%;
      }
    </style>

    <meta http-equiv="Content-Security-Policy" content="default-src * data: gap: https://ssl.gstatic.com 'unsafe-eval'; style-src * 'unsafe-inline'; media-src *">
    <meta name="format-detection" content="telephone=no">
    <meta name="msapplication-tap-highlight" content="no">
    <meta name="viewport" content="user-scalable=no, initial-scale=1, maximum-scale=1, minimum-scale=1, width=device-width">
  </head>
  <body>
    <div id="app"></div>
    <script src="https://d3js.org/d3.v3.min.js" charset="utf-8"></script>
    <script src="js/bundle.js"></script>
  </body>
</html>
* Connection #0 to host localhost left intact

```

**Comment by ghost** [comment]
So the latest finding is. First:

```
→ curl -c cookie -v --data "username=max&password=test" http://localhost:8111/login                                                       [60/7621]
*   Trying ::1...
* Connected to localhost (::1) port 8111 (#0)
> POST /login HTTP/1.1
> Host: localhost:8111
> User-Agent: curl/7.45.0
> Accept: */*
> Content-Length: 26
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 26 out of 26 bytes
< HTTP/1.1 200 OK
< Date: Tue, 10 May 2016 23:33:48 GMT
* Added cookie JSESSIONID="iobgbz8m9qsr129c6ruk0l1t7" for domain localhost, path /, expire 0
< Set-Cookie: JSESSIONID=iobgbz8m9qsr129c6ruk0l1t7;Path=/
< Expires: Thu, 01 Jan 1970 00:00:00 GMT
< Content-Length: 0
< Server: Jetty(9.3.9.M1)
< 
* Connection #0 to host localhost left intact

herakles:  ~/projects/webid-proxy/users  |master ✓|
→ curl -b cookie -H "Accept: text/html, application/xhtml+xml, application/xml;q=0.9,*/*;q=0.8" -v http://localhost:8111/proxy/https://localhost:84
43
*   Trying ::1...
* Connected to localhost (::1) port 8111 (#0)
> GET /proxy/https://localhost:8443 HTTP/1.1
> Host: localhost:8111
> User-Agent: curl/7.45.0
> Cookie: JSESSIONID=iobgbz8m9qsr129c6ruk0l1t7
> Accept: text/html, application/xhtml+xml, application/xml;q=0.9,*/*;q=0.8
> 
< HTTP/1.1 200 OK
< Date: Tue, 10 May 2016 23:34:03 GMT
< Accept-Patch: application/json, application/sparql-update
< Accept-Post: text/turtle, application/json
< Access-Control-Allow-Credentials: true
< Access-Control-Expose-Headers: User, Location, Link, Vary, Last-Modified, WWW-Authenticate, Content-Length, Accept-Patch, Accept-Post, Allow, Upd
ates-Via, Ms-Author-Via
< Access-Control-Max-Age: 1728000
< Allow: OPTIONS, HEAD, GET, PATCH, POST, PUT, MKCOL, DELETE, COPY, MOVE, LOCK, UNLOCK
< Content-Type: text/html
< ETag: "d908beca1f19e567fb9311734ddd7ac1"
< Link: <https://localhost:8443/index.html,meta>; rel="meta", <https://localhost:8443/index.html,acl>; rel="acl"
< Ms-Author-Via: DAV, SPARQL
< Updates-Via: wss://localhost:8443/
< Date: Tue, 10 May 2016 23:34:03 GMT
< Content-Length: 1276
< Server: Jetty(9.3.9.M1)
< 
<!DOCTYPE html>

<html>
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport"
          content="width=device-width, initial-scale=1, user-scalable=no">

    <title>Little Sister</title>

    <link href="https://fonts.googleapis.com/css?family=Roboto:400,300,600"
          rel="stylesheet"
          type="text/css" />
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons"
          rel="stylesheet">

    <style type="text/css">
      * {
        margin: 0;
        padding: 0;
      }
      html, body {
        height: 100%;
      }
      body {
        font-family: Roboto;
      }
      #app {
        width: 100%;
        height: 100%;
      }
    </style>
 <meta http-equiv="Content-Security-Policy" content="default-src * data: gap: https://ssl.gstatic.com 'unsafe-eval'; style-src * 'unsafe-inline'
; media-src *">
    <meta name="format-detection" content="telephone=no">
    <meta name="msapplication-tap-highlight" content="no">
    <meta name="viewport" content="user-scalable=no, initial-scale=1, maximum-scale=1, minimum-scale=1, width=device-width">
  </head>
  <body>
    <div id="app"></div>
    <script src="https://d3js.org/d3.v3.min.js" charset="utf-8"></script>
    <script src="js/bundle.js"></script>
  </body>
</html>

```

Output in Gold Server:

```
[debug] 2016/05/10 23:34:03 server.go:225: 
------ New GET request from 172.17.0.1:43979 ------
[debug] 2016/05/10 23:34:03 auth.go:24: userCookie error: http: named cookie not present
[debug] 2016/05/10 23:34:03 auth.go:44: WebID-TLS error: Get https://max.localhost:8443/profile/card: dial tcp: lookup max.localhost on 192.168.178.1:53: no such host
[debug] 2016/05/10 23:34:03 auth.go:53: Unauthenticated User
[debug] 2016/05/10 23:34:03 server.go:264: 172.17.0.1:43979 requested resource URI: /
[debug] 2016/05/10 23:34:03 server.go:265: 172.17.0.1:43979 requested resource Path: /data/
[debug] 2016/05/10 23:34:03 acl.go:48: Checking accessTo <Read> to https://localhost:8443/ for WebID: 
[debug] 2016/05/10 23:34:03 acl.go:49: Looking for policies in /data/,acl
[debug] 2016/05/10 23:34:03 acl.go:48: Checking defaultForNew <Read> to https://localhost:8443/ for WebID: 
[debug] 2016/05/10 23:34:03 acl.go:49: Looking for policies in /data/,acl
[debug] 2016/05/10 23:34:03 acl.go:183: No ACL policies present - access allowed

```

The strange part is 
`
WebID-TLS error: Get https://max.localhost:8443/profile/card: dial tcp: lookup max.localhost on 192.168.178.1:53: no such host`

Somehow the proxy or some part of gold tries to talk with subdomain (vhost-styled) webid url. Is this something that can be set in the proxy source code?

**Comment by peacekeeper** [comment]
fixed by db5eff157e9d6f518f82cef18d247da03c7c126c
